cql3/selection/selection.cc

```diff
@@ -713,30 +727,7 @@ bool result_set_builder::restrictions_filter::do_filter(const selection& selecti
             }
             break;
         case column_kind::clustering_key: {
-            if (_skip_ck_restrictions) {
-                continue;
-            }
-            const expr::single_column_restrictions_map& clustering_key_restrictions_map =
-                _restrictions->get_single_column_clustering_key_restrictions();
-            auto restr_it = clustering_key_restrictions_map.find(cdef);
-            if (restr_it == clustering_key_restrictions_map.end()) {
-                continue;
-            }
-            if (clustering_key.empty()) {
```

Comment:
You mentioned in the commit message the wish to allow an empty clustering key to support IS NULL. But I'm not sure this applies to clustering keys - keys can never be null!

Reply:
Clustering keys are NULL when the static row is evaluated, and we also support partial clustering prefix as a layover from thrift. Even though we don't support thrift any more, we might have tables that were populated by thrift.